src/api/app/services/package_controller_service/url_generator.rb

```diff
@@ -0,0 +1,58 @@
+module PackageControllerService
+  class URLGenerator
+    def initialize(params = {})
+      @project = params[:project]
+      @repository = params[:repository]
+      @arch = params[:arch]
+      @filename = params[:filename]
+      @user = params[:user]
+      @package = params[:package]
+    end
+
+    def logger
+      Rails.logger
+    end
+
+    def rpm_url
+      get_frontend_url_for(controller: 'build') +
+        "/#{@project.name}/#{@repository.name}/#{@arch}/#{@package.name}/#{@filename}"
+    end
+
+    def download_url_for_file_in_repo
+      download_url = @repository.download_url_for_file(@package.name, @arch, @filename)
+      # return mirror if available
+      return download_url if download_url && file_available?(download_url)
+      # only use API for logged in users if the mirror is not available - return nil otherwise
+      rpm_url unless @user.is_nobody?
+    end
+
+    def file_available?(url, max_redirects = 5)
+      logger.debug "Checking url: #{url}"
+      uri = URI.parse(url)
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.open_timeout = 15
+      http.read_timeout = 15
+      response = http.head uri.path
+      if response.code.to_i == 302 && response['location'] && max_redirects > 0
```

Comment:
[nitpick] Rubocop don't like that `> 0`... :cry:  you should use `.positive?` instead.

Reply:
not my code, i just moved it around.. funny that the ``` rake dev:lint:rubocop``` didn't complain at all..

Reply:
yes, also your code. There are many offenses and that is why Rubocop doesn't complain in this PR (because it is disabled in the TODO file). But this introduces another offense that can't be automatically corrected. This is something that should be avoided :smile: (but it is marked as nitpick anyway)